api/filter_dimension_options.go

```diff
@@ -61,14 +80,35 @@ func (api *FilterAPI) getFilterBlueprintDimensionOptionsHandler(w http.ResponseW
 	log.Event(ctx, "got dimension options for filter blueprint", log.INFO, logData)
 }
 
-func (api *FilterAPI) getFilterBlueprintDimensionOptions(ctx context.Context, filter *models.Filter, dimensionName string) ([]models.PublicDimensionOption, error) {
+// utility function to cut a slice according to the provided offset and limit.
+// limit=0 means no limit, and values higher than the slice length are ignored
+func slice(full []string, offset, limit int) (sliced []string) {
+	end := offset + limit
+	if limit == 0 || end > len(full) {
+		end = len(full)
+	}
 
-	var options []models.PublicDimensionOption
-	dimensionFound := false
-	for _, dimension := range filter.Dimensions {
+	if offset > len(full) {
+		return []string{}
+	}
+	return full[offset:end]
+}
 
+func (api *FilterAPI) getFilterBlueprintDimensionOptions(ctx context.Context, filter *models.Filter, dimensionName string, offset, limit int) (options *models.PublicDimensionOptions, err error) {
+
+	for _, dimension := range filter.Dimensions {
 		if dimension.Name == dimensionName {
-			dimensionFound = true
+
+			options = &models.PublicDimensionOptions{
+				Items:      []*models.PublicDimensionOption{},
+				TotalCount: len(dimension.Options),
+				Offset:     offset,
+				Limit:      limit,
+			}
+
+			// sort alphabetically and cut according to limit and offset
+			sort.Strings(dimension.Options)
+			dimension.Options = slice(dimension.Options, offset, limit)
```

Comment:
don't we want to get the database to do the limit / sorting that is applied here? The database could still return a huge result set that would consume memory when it's never used.

Reply:
yes, I tried to do that but I think we would need to split the filters into separate filters and options collections? (similar to dataset api), or alternatively, do you know if it's possible to query parts of a document? (if it's possible we could query everything except the options, and then query only the options that we need - I haven't seen if this is possible)

Reply:
It feels like we should separate the options, as there seems to be a few cases where they are not wanted. I am not sure if that it worth doing right now though. I will go ahead and approve, and we can see if we need to go further.